### PR TITLE
Invisible lurker pushing

### PR DESCRIPTION
**Pull request by morrowwolf**
# About the pull request

This PR makes it so when you collide with a human as an invisible lurker you will become visible.

# Explain why it's good for the game

Being pushed by a fast and barely visible xeno is super cheesy.

# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog

:cl: Morrow
balance: Lurkers now lose their invisibility when they run into a person
/:cl:

### CODE REVIEW

**Comment by cm13live** [comment]
This pull request has been mentioned on **CM-SS13**. Please provide feedback here:

https://forum.cm-ss13.com/t/whats-with-all-the-xeno-overhauls-lately/2753/103

**Comment by TheGamerdk** [comment]
Couldn’t you just make invisible lurkers unable to push people

**Comment by TheGamerdk** [comment]
> Couldn’t you just make invisible lurkers unable to push people

How is asking this off topic

**Comment by ThiccFucc** [comment]
:1984:

File: code/modules/mob/living/carbon/xenomorph/castes/Lurker.dm [hunk 1]
```diff
@@ -151,3 +151,14 @@
...
 	var/invis_message = (invis_start_time == -1) ? "N/A" : "[(invis_duration-(world.time - invis_start_time))/10] seconds."
 	. += "Invisibility Time Left: [invis_message]"
+
+/datum/behavior_delegate/lurker_base/on_collide(atom/movable/movable_atom)
+	. = ..()
+
+	if(!ishuman(movable_atom))
+		return
+
+	if(bound_xeno && bound_xeno.stealth)
```
Comment (harryob) [inline code/modules/mob/living/carbon/xenomorph/castes/Lurker.dm:161]:
```suggestion
	if(!bound_xeno || !bound_xeno.stealth)
		return

```

File: code/modules/mob/living/carbon/xenomorph/castes/Lurker.dm [hunk 2]
```diff
@@ -151,3 +151,14 @@
...
+		var/datum/action/xeno_action/onclick/lurker_invisibility/lurker_invisibility_action = get_xeno_action_by_type(bound_xeno, /datum/action/xeno_action/onclick/lurker_invisibility)
```
Comment (harryob) [inline code/modules/mob/living/carbon/xenomorph/castes/Lurker.dm:162]:
```suggestion
	 var/datum/action/xeno_action/onclick/lurker_invisibility/lurker_invisibility_action = get_xeno_action_by_type(bound_xeno, /datum/action/xeno_action/onclick/lurker_invisibility)
```

File: code/modules/mob/living/carbon/xenomorph/castes/Lurker.dm [hunk 3]
```diff
@@ -151,3 +151,14 @@
...
+		if(lurker_invisibility_action)
```
Comment (harryob) [inline code/modules/mob/living/carbon/xenomorph/castes/Lurker.dm:163]:
```suggestion
	if(!lurker_invisibility_action)
		return

```

File: code/modules/mob/living/carbon/xenomorph/castes/Lurker.dm [hunk 4]
```diff
@@ -151,3 +151,14 @@
...
+			lurker_invisibility_action.invisibility_off()
```
Comment (harryob) [inline code/modules/mob/living/carbon/xenomorph/castes/Lurker.dm:164]:
```suggestion
	lurker_invisibility_action.invisibility_off()
```

**Review by harryob — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/xenomorph/castes/Lurker.dm [hunk 5]
```diff
@@ -151,3 +151,18 @@
...
+/datum/behavior_delegate/lurker_base/on_collide(atom/movable/movable_atom)
+	. = ..()
+
+	if(!ishuman(movable_atom))
+		return
+
+	if(!bound_xeno || !bound_xeno.stealth)
+		return
+
+	 var/datum/action/xeno_action/onclick/lurker_invisibility/lurker_invisibility_action = get_xeno_action_by_type(bound_xeno, /datum/action/xeno_action/onclick/lurker_invisibility)
```
Comment (BeagleGaming1) [inline code/modules/mob/living/carbon/xenomorph/castes/Lurker.dm:164]:
```suggestion
	 var/datum/action/xeno_action/onclick/lurker_invisibility/lurker_invisibility_action = get_xeno_action_by_type(bound_xeno, /datum/action/xeno_action/onclick/lurker_invisibility_action)
```

File: code/modules/mob/living/carbon/xenomorph/castes/Lurker.dm [hunk 6]
```diff
@@ -151,3 +151,18 @@
...
+/datum/behavior_delegate/lurker_base/on_collide(atom/movable/movable_atom)
+	. = ..()
+
+	if(!ishuman(movable_atom))
+		return
+
+	if(!bound_xeno || !bound_xeno.stealth)
+		return
+
+	 var/datum/action/xeno_action/onclick/lurker_invisibility/lurker_invisibility_action = get_xeno_action_by_type(bound_xeno, /datum/action/xeno_action/onclick/lurker_invisibility_action)
```
Comment (BeagleGaming1) [inline code/modules/mob/living/carbon/xenomorph/castes/Lurker.dm:164]:
```suggestion
	var/datum/action/xeno_action/onclick/lurker_invisibility/lurker_invisibility_action = get_xeno_action_by_type(bound_xeno, /datum/action/xeno_action/onclick/lurker_invisibility_action)
```
Real

File: code/modules/mob/living/carbon/xenomorph/castes/Lurker.dm [hunk 7]
```diff
@@ -151,3 +151,18 @@
...
+/datum/behavior_delegate/lurker_base/on_collide(atom/movable/movable_atom)
+	. = ..()
+
+	if(!ishuman(movable_atom))
+		return
+
+	if(!bound_xeno || !bound_xeno.stealth)
+		return
+
+	var/datum/action/xeno_action/onclick/lurker_invisibility/lurker_invisibility_action = get_xeno_action_by_type(bound_xeno, /datum/action/xeno_action/onclick/lurker_invisibility_action)
```
Comment (BeagleGaming1) [inline code/modules/mob/living/carbon/xenomorph/castes/Lurker.dm:164]:
```suggestion
	var/datum/action/xeno_action/onclick/lurker_invisibility/lurker_invisibility_action = get_xeno_action_by_type(bound_xeno, /datum/action/xeno_action/onclick/lurker_invisibility)
```

**Review by BeagleGaming1 — APPROVED** [review]
Code seems good, still needs balance approval.

**Pull request merged**